require.js

```diff
@@ -49,10 +49,29 @@
     }
 
     var globalEval = eval; // reassigning causes eval to not use lexical scope.
+    var ArrayPush = Array.prototype.push;
 
     // Non-CommonJS speced extensions should be marked with an "// EXTENSION"
     // comment.
 
+
+	var _Module = function _Module() {};
+	_Module.prototype.id = null;
+	_Module.prototype.display = null;
+	_Module.prototype.require = null;
+	_Module.prototype.factory = void 0;
+	_Module.prototype.exports = void 0;
+	_Module.prototype.redirect = void 0;
+	_Module.prototype.location = null;
+	_Module.prototype.directory = null;
+	_Module.prototype.injected = false;
+	_Module.prototype.mappingRedirect = void 0;
+	_Module.prototype.type = null;
+	_Module.prototype.text = void 0;
+	_Module.prototype.dependees = null;
+	_Module.prototype.extraDependencies = void 0;
+	_Module.prototype.uuid = null;
```

Comment:
The prototype approach doesn’t help any engine. The constructor approach only helps V8.